src/server/tokenization.ts

```diff
@@ -0,0 +1,115 @@
+import * as use from '@tensorflow-models/universal-sentence-encoder';
+import * as punctuations from 'unicode-12.1.0/General_Category/Punctuation/code-points';
+import * as nsmarks from 'unicode-12.1.0/General_Category/Nonspacing_Mark/code-points';
+import * as ctlchar from 'unicode-12.1.0/General_Category/Control/code-points';
+import * as spcsep from 'unicode-12.1.0/General_Category/Space_Separator/code-points';
+
+function is_punctuation(cp: number):boolean {
+  // Checks a cp is a punctuation character or not.
+  return (punctuations.indexOf(cp) == -1) ? false:true;
+}
+
+function run_strip_accents(text: string){
+  // strips accents from text
+  const mn: number[] = nsmarks;
+  text = text.normalize("NFD");
+  mn.forEach(cp => {text = text.replace(String.fromCodePoint(cp), "")});
+  return text;
+}
+
+function is_white_space(cp: number): boolean{
+  // \t, \n, and \r are technically contorl characters but we treat them
+  // as whitespace since they are generally considered as such.
+  if (cp == 32 || cp == 9 || cp == 10 || cp == 13) {
+    return true;
+  }
+  if (spcsep.indexOf(cp) != -1){
+    return true;
+  }
+  return false;
+}
+
+function is_constrol(cp: number): boolean{
+  // "\t" "\n" "\r" are technically control characters but we count them as whitespace
+  // characters.
+  if (cp == 9 || cp == 10 || cp == 13) {
+    return false;
+  }
+  if (ctlchar.indexOf(cp) != -1){
+    return true;
+  }
+  return false;
+}
+
+function run_split_on_punc(text: string){
+  // Splits punctuation with a space on a piece of text
+  // e.g. "abc?" -> "abc ?"
+  // "abc?def" -> "abc ? def"
+  const output = [];
+  let preCodePoint = -1;
+  for (let i = 0; i < text.length; i++){
+    const codePoint = text.charCodeAt(i);
+    let postCodePonit = text.charCodeAt(i+1);
+    if (is_punctuation(codePoint)) {
+      // is punctuations
+      if (preCodePoint !== 32) {
+        output.push(32);
+        output.push(codePoint);
+      }
+      if (isNaN(postCodePonit) == false && postCodePonit != 32){
+        output.push(32);
+      }
+    } else {
+      output.push(codePoint);
+    }
+    preCodePoint = codePoint;
+  }
+  return String.fromCodePoint(...output);
+}
+
+function clean_text(text: string){
+  //Performs invalid character removal and whitespace cleanup on text.
+  const output = [];
+  for (let i = 0; i < text.length; i++){
+    let cp = text.charCodeAt(i);
+    if (cp == 0 || cp == 65533 || is_constrol(cp)){
+      continue;
+    }
+    if (is_white_space(cp)){
+      output.push(32);
+    }
+    else {
+      output.push(cp);
+    }
+  }
+  return String.fromCharCode(...output);
+}
+
+export default class WordPieceTokenizer{
+  //Runs basic tokenization (punctuation splitting, lower casing, etc.).
+  private do_lower_case: boolean;
```

Comment:
nit: Camel Case

Reply:
It seems TFJS code uses all Camel Case, so maybe we should be consistent and use all Camel Case.

Reply:
All done!